File: Universal Windows Platform Solution/Phantom of the West/Phantom of the West/Visual Novel/Chapters/Chapter 0/Chapter0.cs
using Phantom_of_the_West.User_Interface.Story_View;
using System.Collections.Generic;

namespace Phantom_of_the_West.Visual_Novel.Chapters.Chapter_0
{
	internal class Chapter0 : Chapter
	{
		private const string leftArrow = "⬅︎";
		private const string ok = "OK";
		private const string rightArrow = "➡︎";
		private const string menuIcon = "≣";

		private static Dictionary<int, StoryChoiceListDelegate> dictionary = new Dictionary<int, StoryChoiceListDelegate>
		{
			{1, Ch0Welcome },
			{2, Ch0POTWSimple },
			{3, Ch0MultipleChoices },
			{4, Ch0FirstChoice },
			{5, Ch0Options }
		};

		internal Chapter0() : base(dictionary)
		{
		}

		private static StoryChoiceList Ch0Welcome()
		{
			string localizedString = StoryChoice.GetLocalizedString("ch0Welcome");
			string formattedLS = string.Format(localizedString, ok);
			return StoryChoiceList.FromResource("Black.png", new IStoryChoice[] { new StoryChoice(formattedLS, 2) });
		}

		private static StoryChoiceList Ch0POTWSimple()
		{
			string localizedString = StoryChoice.GetLocalizedString("ch0POTWSimple");
			string formattedLS = string.Format(localizedString, ok);
			return StoryChoiceList.FromResource("Black.png", new IStoryChoice[] { new StoryChoice(formattedLS, 3) });
		}

		private static StoryChoiceList Ch0MultipleChoices()
		{
			string localizedString = StoryChoice.GetLocalizedString("ch0MultipleChoices");
			string formattedLS = string.Format(localizedString, ok);
			return StoryChoiceList.FromResource("Black.png", new IStoryChoice[] { new StoryChoice(formattedLS, 4) });
		}

		private static StoryChoiceList Ch0FirstChoice()
		{
			string localizedString1 = StoryChoice.GetLocalizedString("ch0FirstChoice");
			string localizedString2 = StoryChoice.GetLocalizedString("ch0SecondChoice");
			string formattedLS1 = string.Format(localizedString1, rightArrow, ok);
			string formattedLS2 = string.Format(localizedString2, leftArrow, ok);
			return StoryChoiceList.FromResource("Black.png", new IStoryChoice[] { (new StoryChoice(formattedLS1, 5)), (new StoryChoice(formattedLS2, 5)) });
		}

		private static StoryChoiceList Ch0Options()
		{
			string localizedString = StoryChoice.GetLocalizedString("ch0Options");
			string formattedLS = string.Format(localizedString, menuIcon, ok);
			return StoryChoiceList.FromResource("Black.png", new IStoryChoice[] { new StoryChoice(formattedLS, 1001) });
		}
	}
}

File: Android Project/app/src/main/java/com/christopherraleigh/phantomofthewest/visual_novel/chapters/chapter_0/Chapter0.java
package com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_0;

import com.christopherraleigh.phantomofthewest.R;
import com.christopherraleigh.phantomofthewest.visual_novel.IStoryChoice;
import com.christopherraleigh.phantomofthewest.visual_novel.IStoryChoiceList;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.Chapter;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.StoryChoiceList;

import java.util.Arrays;

/**
 * Created by christopherraleigh on 2016-04-19.
 */
public class Chapter0 extends Chapter {
    private static StoryChoiceList ch0Welcome = new StoryChoiceList(R.drawable.black,
            Arrays.asList(new IStoryChoice[]{new Welcome()}));

    private static StoryChoiceList ch0POTWSimple = new StoryChoiceList(R.drawable.black,
            Arrays.asList(new IStoryChoice[]{new POTWSimple()}));

    private static StoryChoiceList ch0MultipleChoices = new StoryChoiceList(R.drawable.black,
            Arrays.asList(new IStoryChoice[]{new MultipleChoices()}));

    private static StoryChoiceList ch0FirstChoice = new StoryChoiceList(R.drawable.black,
            Arrays.asList(new IStoryChoice[]{
                    new FirstChoice(),
                    new SecondChoice()
            }));

    private static StoryChoiceList ch0Options = new StoryChoiceList(R.drawable.black,
            Arrays.asList(new IStoryChoice[]{new Options()}));

    @Override
    public IStoryChoiceList goToState(int id) {
        switch (id) {
            case 1:
                return ch0Welcome;
            case 2:
                return ch0POTWSimple;
            case 3:
                return ch0MultipleChoices;
            case 4:
                return ch0FirstChoice;
            case 5:
                return ch0Options;
        }
        return null;
    }
}
